### PR TITLE
Ms 3473 - Rewrite

### PR DESCRIPTION
**Pull request by pontus-eliason**
Rewritten Java SDK to be typesafe and easier to work with.

Not backwards compatible with previous SDK.

@trustly/ceres

### CODE REVIEW

File: src/test/java/com/trustly/api/NotificationsTest.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+          responseString.set(value);
+        }
+      };
+
+      TrustlyApiClientExtensions.handleNotificationRequest(is, responder);
+
+      Assertions.assertEquals(200, status.get());
+      Assertions.assertNotNull(responseString.get());
+    }
+  }
```
Comment (BergJonatan) [inline src/test/java/com/trustly/api/NotificationsTest.java:176]: Why have a specific test for `Cancel` when it is included in the first test?

File: src/test/java/com/trustly/api/RequestsTest.java [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+import java.util.UUID;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * These test will only work if a local environment exists. At the USER_HOME root these files is needed: trustly_client_username.txt - your
+ * username trustly_client_password.txt - your password trustly_client_public.pem - your public key trustly_client_public.pem - your private
+ * key
+ * <p>
+ * If those files exist and is correct you are able to make requests to Trustlys test environment
```
Comment (BergJonatan) [inline src/test/java/com/trustly/api/RequestsTest.java:39]: are*

File: src/test/java/com/trustly/api/RequestsTest.java [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+        .notificationUrl("https://fake.test.notification.trustly.com")
+        .endUserId("pontus.eliason@trustly.com")
+        .messageId(UUID.randomUUID().toString())
+        .attributes(
+          DepositRequestDataAttributes.builder()
+            .currency("EUR")
+            .amount("100.00")
+            .firstname("John")
+            .lastname("Doe")
+            .email("pontus.eliason@trustly.com")
```
Comment (BergJonatan) [inline src/test/java/com/trustly/api/RequestsTest.java:63]: Is there a reason to not use John.Doe@trustly.com?

**Review by BergJonatan — APPROVED** [review]

File: src/test/java/com/trustly/api/NotificationsTest.java (hunk 1 above)
Comment (pontus-eliason) [reply src/test/java/com/trustly/api/NotificationsTest.java:176]: That one is with signature verification, the others are not

File: src/test/java/com/trustly/api/RequestsTest.java (hunk 3 above)
Comment (pontus-eliason) [reply src/test/java/com/trustly/api/RequestsTest.java:63]: Nope :) Will fix

**Pull request merged**